src/components/autoSuggest/index.js

```diff
@@ -0,0 +1,213 @@
+import resultList from './resultList';
+import routes from './../../constants/routes';
+import keyCodes from './../../constants/keyCodes';
+import { visitAndSaveSearch } from './../search/keyAction';
+import mockSearchResults from './searchResults.mock';
+
+let searchResults = mockSearchResults;
+class AutoSuggest extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.submitSearch = this.submitSearch.bind(this);
+
+    searchResults = this.props.results || searchResults;
+
+    this.delegatesPropsMap = {
+      uniqueKey: 'address',
+      redirectPath: entity => `${routes.accounts.pathPrefix}${routes.accounts.path}/${entity.address}`,
+      keyHeader: 'username',
+      keyValue: entity => (<span>{entity.rank}</span>),
+      i18Header: this.props.t('Delegate'),
+      i18Value: this.props.t('Rank'),
+    };
+
+    this.addressesPropsMap = {
+      uniqueKey: 'address',
+      redirectPath: entity => `${routes.accounts.pathPrefix}${routes.accounts.path}/${entity.address}`,
+      keyHeader: 'address',
+      keyValue: entity => (<span><LiskAmount val={entity.balance}/> LSK</span>),
+      i18Header: this.props.t('Address'),
+      i18Value: this.props.t('Balance'),
+    };
+
+    this.transactionsPropsMap = {
+      uniqueKey: 'id',
+      redirectPath: entity => `${routes.transactions.pathPrefix}${routes.transactions.path}/${entity.id}`,
+      keyHeader: 'id',
+      keyValue: entity => (<span>{entity.height}</span>),
+      i18Header: this.props.t('Transaction'),
+      i18Value: this.props.t('Height'),
+    };
+
+    this.commonProps = {
+      submitSearch: this.submitSearch,
+      selectedRowProps: {
+        ref: this.setSelectedRow.bind(this),
+      },
+    };
+
+    let resultsLength = 0;
+    Object.keys(searchResults).map((resultKey) => {
+      resultsLength += searchResults[resultKey].length;
+      return resultsLength;
+    });
+
+    this.selectedRow = null;
+
+    this.state = {
+      show: false,
+      value: '',
+      selectedIdx: 0,
+      resultsLength,
+    };
+  }
+
+  setSelectedRow(el) {
+    this.selectedRow = el;
+  }
+
+  submitSearch(urlSearch) {
+    this.resetSearch();
+    this.inputRef.blur();
+    if (!urlSearch) {
+      this.selectedRow.click();
+      return;
+    }
+    this.props.history.push(urlSearch);
+  }
+
+  search(searchTerm) {
+    if (searchTerm !== '') {
+      this.setState({ show: true });
+    } else {
+      this.setState({ show: false });
+    }
+    this.setState({ value: searchTerm });
+  }
+
+  handleArrowDown() {
+    let currentIdx = this.state.selectedIdx;
+    currentIdx = (currentIdx === this.resultsLength) ? this.resultsLength : currentIdx += 1;
+    this.setState({ selectedIdx: currentIdx });
+  }
+
+  handleArrowUp() {
+    let currentIdx = this.state.selectedIdx;
+    currentIdx = (currentIdx === 0) ? 0 : currentIdx -= 1;
+    this.setState({ selectedIdx: currentIdx });
+  }
+
+  handleBlur() {
+    this.closeDropdown();
+  }
```

Comment:
You can use directly `this.closeDropdown` and remove this method.

Reply:
done